cvm-attestation/src/ImdsClient.py

```diff
@@ -31,105 +32,108 @@ class ImdsClient:
+      return evidence_json['quote']
+    except json.JSONDecodeError as e:
+      self.log.error("Failed to decode TD quote JSON", exc_info=True)
+      raise TDQuoteException(f"JSON decoding error: {e}") from e
 
   def get_vcek_certificate(self):
-    # setup imds request
-    headers = {
-      'Content-Type': 'application/json',
-      'Metadata': 'true'
-    }
-
-    # Request Vcek Certificate with 5 retries, increasing delay by 2x each request
-    delay = 1
-    max_retries = 5
-    retries = 0
-    while retries < max_retries:
-      try:
-        self.log.info("Starting vcek certificate request")
-        response = requests.get(
-          THIM_ENDPOINT,
-          headers = headers
-        )
-
-        if response.status_code == 200:
-          data_json = json.loads(response.text)
-          self.log.info(f"Received certificate successfully, TCB version: {data_json['tcbm']}")
-          cert = data_json['vcekCert']
-          chain = data_json['certificateChain']
-          cert_chain = cert + chain
-          cert_chain = bytearray(cert_chain.encode('utf-8'))
-
-          return cert_chain
-        else:
-          self.log.error('Failed to get vcek certificate')
-          self.log.error(f'response: {response.text}')
-
-          retries += 1
-          if retries >= max_retries:
-            raise VcekCertException(f'Error {response.status_code}: {response.text}')
-
-          self.log.info(f"Retrying in {delay} seconds... Attempt {retries}")
-          time.sleep(delay)
-          delay *= 2
-      except requests.exceptions.RequestException as e:
-        self.log.error("Exception while fetching vcek certificate", exc_info=True)
+    """
+    Get the VCEK certificate from the IMDS endpoint.
+
+    Returns:
+      The VCEK certificate received from the IMDS endpoint.
+    """
+
+    response = self._send_request_with_retries(
+      method='get',
+      url=THIM_ENDPOINT,
+      headers=METADATA_HEADERS,
+      delay_strategy='exponential',
+      exception_class=VcekCertException
+    )
+
+    try:
+      data_json = response.json()
+      self.log.info(f"Received VCEK certificate, TCB version: {data_json.get('tcbm')}")
+      cert_chain = data_json['vcekCert'] + data_json['certificateChain']
+      return bytearray(cert_chain.encode('utf-8'))
+    except json.JSONDecodeError as e:
+      self.log.error("Failed to decode VCEK certificate JSON", exc_info=True)
+      raise VcekCertException(f"JSON decoding error: {e}") from e
+
+def get_region_from_compute_metadata(self):
+  """
+  Get the region from the compute metadata.
+
+  Returns:
+    The region string if available, otherwise None.
+  """
+  try:
+    response = self._send_request_with_retries(
+      method='get',
+      url=COMPUTE_METADATA_URL,
+      headers=METADATA_HEADERS,
+      exception_class=TDQuoteException  # Or define a specific exception if needed
+    )
+
+    metadata = response.json()
+    return metadata.get('compute', {}).get('location')
```

Comment:
IMDS supports this [Route Parameter](https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=linux#route-parameters). So, you can do
`url="http://169.254.169.254/metadata/instance/compute/location?api-version=2021-01-01&format=text"` to get the location value directly.

Reply:
Thanks, I added this on my latest commit!! This is a great find!